client/client.go

```diff
@@ -68,6 +69,10 @@ func (c *Client) DoAndUnmarshal(ctx context.Context, r *Request, v interface{})
 		return err
 	}
 
+	if response.Close {
```

Comment:
`http.Response.Close` represents if the underlying transport protocol should be closed or kept open.

In this case I think the body.Close should be done in the Unmarshal function? Since it is there were we consume the stream.

Reply:
My idea was to keep it close to the actual read so when you touch that code and refactor things around you understand that you are also supposed to handle closing of the body.

Might be a good idea to just add it to both, as closing a closed body is a no-op. Too make sure it survives the next guy poking around.